traversal/predicate/PredicateArgument.java

```diff
@@ -36,7 +36,7 @@ public abstract class PredicateArgument {
 
     private final String symbol;
 
-    protected PredicateArgument(String symbol) {
+    PredicateArgument(String symbol) {
```

Comment:
since the `Predicate` is more generally useful now, we don't need to expose any methods in this class to be accessed publicly